### PR TITLE
refactor zq packaging and add some fit and finish

### PR DESCRIPTION
**Pull request by mccanne**
This commit refactors zq so we can build it in an outside repo.
While we were at it, we cleaned up the online help and changed
the -D flag to -split, and added queryflags.  We also renamed
emitter.Dir to emitter.Split and simplified the logic.

### CODE REVIEW

File: cmd/zq/main.go [hunk 1]
```diff
@@ -1,154 +1,14 @@
...
-	"github.com/brimdata/zed/cli/outputflags"
-	"github.com/brimdata/zed/cli/procflags"
-	zedquery "github.com/brimdata/zed/cmd/zed/query"
-	"github.com/brimdata/zed/cmd/zed/root"
-	"github.com/brimdata/zed/driver"
-	"github.com/brimdata/zed/pkg/charm"
-	"github.com/brimdata/zed/pkg/rlimit"
-	"github.com/brimdata/zed/pkg/storage"
-	"github.com/brimdata/zed/zio"
+	"github.com/brimdata/zed/cli/zq"
```
Comment (mattnibs) [inline cmd/zq/main.go:7]: Why not just put the Command here if zq is the command is only used here?

**Review by mattnibs — APPROVED** [review]

File: cli/zq/command.go [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+so you can mix and match input types.  If multiple files are concatenated
+into a stream and presented as standard input, the files must all be of the
+same type as the beginning of stream will determine the format.
+
+Output is sent to standard output unless an output file is specified with -o.
+Some output formats like Parquet are based on schemas and require all
+data in the output to conform to the same schema.  To handle this, you can
+either fuse the data into a union of all the record types present
+(presuming all the output values are records) or you can specify the -split
+flag to indicate that a directory to place separate output files for each
```
Comment (nwt) [inline cli/zq/command.go:56]:
```suggestion
flag to indicate a destination directory for separate output files for each
```

File: cli/zq/command.go [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+output type.  This flag may be used in combination with -o, which
+provides the prefix for the file path, e.g.,
+
+  zq -f parquet -split out -o example-output input.zng
+
+When writing to stdout and stdout is a terminal, the default output format is ZSON.
+Otherwise, the default format is binary ZNG.  In either case, the default
+may be overridden with -f.
```
Comment (nwt) [inline cli/zq/command.go:64]:
```suggestion
may be overridden with -f, -z, or -Z.
```

File: cli/zq/command.go [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+
+If the first argument is a path to a valid file rather than a Zed query,
+then the Zed query is assumed to be "*", i.e., match and output all
+of the input.  If the first argument is both a valid Zed query
+and an existing file, then the file overrides.
+
+The Zed query text may include source files using -I, which is particularly
+convenient when a large, complex query spans multiple lines.  In this case,
+these source files are concatenated together along with the command-line query text
+in the order appearing on the command-line.
```
Comment (nwt) [inline cli/zq/command.go:78]: Usually not hyphen
```suggestion
in the order appearing on the command line.
```

File: cli/zq/command.go [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+"zq" is a command-line tool for processing data in diverse input formats,
+providing search, analytics, and extensive transormations using
+the Zed query language.
+A query typically applies boolean logic or keyword search to filter
+the input then performs analytics and/or transformations to the filtered stream.
+Output is written to one or more files or to standard output.
+
+A Zed query is comprised of one or more operators interconnected
+into a pipeline using the Unix pipe charcter "|".
+See https://github.com/brimdata/docs/language for details.
```
Comment (nwt) [inline cli/zq/command.go:37]:
```suggestion
See https://github.com/brimdata/zed/tree/main/docs/language for details.
```

File: cli/zq/command.go [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+
+  zq -f zng input.json > fast.zng
+  zq <query> fast.zng
+  ...
+
+Please see the zq or Zed github repositories for more information:
+
+https://github.com/brimdata/zq
+
+https://github.com/brimdata/zed
```
Comment (nwt) [inline cli/zq/command.go:92]:
```suggestion
Please see https://github.com/brimdata/zq and https://github.com/brimdata/zed for more information.
```

File: cli/zq/command.go [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+then the Zed query is assumed to be "*", i.e., match and output all
+of the input.  If the first argument is both a valid Zed query
+and an existing file, then the file overrides.
+
+The Zed query text may include source files using -I, which is particularly
+convenient when a large, complex query spans multiple lines.  In this case,
+these source files are concatenated together along with the command-line query text
+in the order appearing on the command-line.
+
+The zq engine processes data natively in Zed so if you intend to run
```
Comment (nwt) [inline cli/zq/command.go:80]:
```suggestion
The "zq" engine processes data natively in Zed so if you intend to run
```

File: cli/zq/command.go [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+many queries over the same data, you will see substantil performance gains
```
Comment (nwt) [inline cli/zq/command.go:81]:
```suggestion
many queries over the same data, you will see substantial performance gains
```

File: zio/emitter/split.go [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+	"github.com/brimdata/zed"
+	"github.com/brimdata/zed/pkg/storage"
+	"github.com/brimdata/zed/zio"
+	"github.com/brimdata/zed/zio/anyio"
+)
+
+var (
+	ErrNoPath = errors.New("no _path field in zng record")
+)
+
```
Comment (nwt) [inline zio/emitter/split.go:18]: Delete this while you're here?
```suggestion
```

**Review by nwt — APPROVED** [review]

File: cli/zq/command.go (hunk 6 above)
Comment (mccanne) [reply cli/zq/command.go:92]: These usually get formatted bad in charm but I'll make the change.

File: cmd/zq/main.go (hunk 1 above)
Comment (mccanne) [reply cmd/zq/main.go:7]: The point is to be able to do an install from either brimdata/zed or brimdata/zq and the pending repo brimdata/zq will have a main that looks the same as cmd/zq/main.go.

**Pull request merged**